scripts/prep-spd-roster.py

```diff
@@ -1,49 +1,80 @@
 import pandas as pd
 import click
 
-from pandas.core.frame import DataFrame
+from pandas import DataFrame
 
 
 @click.command()
-@click.option('-d', '--date', required=True, type=str, prompt=True, help='The date of the roster in the format YYYY-MM-DD.')
-@click.option('-i', '--in-csv', required=True, type=str, prompt=True, help='Path to the source file. Can be relative to current directory.')
-@click.option('-o', '--out-csv', required=True, type=str, prompt=True, help='Path to save the resulting CSV in. Can be relative to current directory.')
+@click.option(
+    "-d",
+    "--date",
+    required=True,
+    type=str,
+    prompt=True,
+    help="The date of the roster in the format YYYY-MM-DD.",
+)
+@click.option(
+    "-i",
+    "--in-csv",
+    required=True,
+    type=str,
+    prompt=True,
+    help="Path to the source file. Can be relative to current directory.",
+)
+@click.option(
+    "-o",
+    "--out-csv",
+    required=True,
+    type=str,
+    prompt=True,
+    help="Path to save the resulting CSV in. Can be relative to current directory.",
+)
 def main(date, in_csv, out_csv):
     # Import CSV passed to script
     df = pd.read_csv(in_csv)
 
     # Rename columns to better formatting
     column_names = {
-        'Name': 'full_name',
-        'Badge_Num': 'badge',
-        'Title_Description': 'title',
-        'Unit': 'unit',
-        'Unit_Description': 'unit_description'
+        "Name": "full_name",
+        "Badge_Num": "badge",
+        "Serial": "badge",
+        "Title_Description": "title",
+        "Title Description": "title",
+        "Unit": "unit",
+        "Unit Description": "unit_description",
+        "Unit_Description": "unit_description",
     }
-    df.rename(columns=column_names, inplace=True)
+    df = df.rename(columns=column_names)
 
     # Strip out unwanted whitespace from any string fields in the DataFrame
     df = df.applymap(lambda x: x.strip() if isinstance(x, str) else x)
 
     # Split the names out
-    df['first_name'] = df["full_name"].str.split(",").str[1].str.strip().str.split(" ").str[0]
-    df['middle_name'] = df.apply(lambda x: middle_name(x), axis=1)
-    df['last_name'] = df["full_name"].str.split(",").str[0]
+    names = df["full_name"].str.split(",", expand=True)
+    # Last name is predictably before the comma, so we can set it outright
+    df["last_name"] = names[0]
+    # Assumption: first name will always be the text before the first space
+    # We can set first name to be the first split based off that
+    not_last_names = names[1].str.split(n=1, expand=True)
+    df["first_name"] = not_last_names[0]
+    # Next we have to split apart the middle name and the suffix
+    # Sometimes there some weird "K_Jr" bullshit for the middle name we have to deal with
+    # This creates a dataframe with 3 new columns: 1) middle bits, 2) suffix, 3) blank because of the split
+    middle_bits = not_last_names[1].str.split(
+        r"((?:Jr|I[iI]|I[iI][iI]|I[vV])\.?$)", expand=True

```

Comment:
You can simplify this to:
`r"(?i)((?:jr|ii|iii|iv)\.?$)", expand=True`

The `(?i)` indicates that from there until either the next `(?-i)` or the end of the regex matching will be case insensitive. Wasn't sure whether you would prefer to do all lower or all upper for the regex for readability.